src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/DefaultJavaScriptEncoder.cs

```diff
@@ -87,21 +88,30 @@ public override unsafe int FindFirstCharacterToEncodeUtf8(ReadOnlySpan<byte> utf
                 int idx = 0;
 
 #if NETCOREAPP
-                if (Sse2.IsSupported)
+                if (Sse2.IsSupported || AdvSimd.Arm64.IsSupported)
                 {
                     sbyte* startingAddress = (sbyte*)ptr;
                     while (utf8Text.Length - 16 >= idx)
                     {
                         Debug.Assert(startingAddress >= ptr && startingAddress <= (ptr + utf8Text.Length - 16));
 
-                        // Load the next 16 bytes.
-                        Vector128<sbyte> sourceValue = Sse2.LoadVector128(startingAddress);
+                        bool containsNonAsciiBytes;
 
-                        // Check for ASCII text. Any byte that's not in the ASCII range will already be negative when
-                        // casted to signed byte.
-                        int index = Sse2.MoveMask(sourceValue);
+                        // Load the next 16 bytes, and check for ASCII text.
+                        // Any byte that's not in the ASCII range will already be negative when casted to signed byte.
+                        if (Sse2.IsSupported)
+                        {
+                            Vector128<sbyte> sourceValue = Sse2.LoadVector128(startingAddress);
+                            containsNonAsciiBytes = Sse2.MoveMask(sourceValue) != 0;
+                        }
+                        else
+                        {
+                            Debug.Assert(AdvSimd.Arm64.IsSupported);
+                            Vector128<sbyte> sourceValue = AdvSimd.LoadVector128(startingAddress);
+                            containsNonAsciiBytes = AdvSimd.Arm64.MinAcross(sourceValue).ToScalar() < 0;
```

Comment:
Why did you decide to use `AdvSimd.Arm64.MinAcross(sourceValue).ToScalar()` here instead of your new method `AdvSimdHelper.MoveMask`?

Reply:
The actual bit positions are not required here, so using this instead because it's just one instruction.